x-pack/plugins/ml/public/application/components/ml_embedded_map/ml_embedded_map.tsx

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { useEffect, useRef, useState } from 'react';
+import uuid from 'uuid';
+import { LayerDescriptor } from '../../../../../maps/common/descriptor_types';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { MapEmbeddable, MapEmbeddableInput } from '../../../../../maps/public/embeddable';
+import { MAP_SAVED_OBJECT_TYPE, RenderTooltipContentParams } from '../../../../../maps/public';
+import {
+  ErrorEmbeddable,
+  isErrorEmbeddable,
+  ViewMode,
+} from '../../../../../../../src/plugins/embeddable/public';
+import { useMlKibana } from '../../contexts/kibana';
+
+export function MlEmbeddedMapComponent({
+  layerList,
+  mapEmbeddableInput,
+  renderTooltipContent,
+}: {
+  layerList: LayerDescriptor[];
+  mapEmbeddableInput?: MapEmbeddableInput;
+  renderTooltipContent?: (params: RenderTooltipContentParams) => JSX.Element;
+}) {
+  const [embeddable, setEmbeddable] = useState<MapEmbeddable | ErrorEmbeddable | undefined>();
+
+  const embeddableRoot: React.RefObject<HTMLDivElement> = useRef<HTMLDivElement>(null);
+  const baseLayers = useRef<LayerDescriptor[]>();
+
+  const {
+    services: { embeddable: embeddablePlugin, maps: mapsPlugin },
+  } = useMlKibana();
+
+  if (!embeddablePlugin) {
+    throw new Error('Embeddable start plugin not found');
+  }
+  if (!mapsPlugin) {
+    throw new Error('Maps start plugin not found');
+  }
```

Comment:
do we need to throw errors? can we just check if it isn't available and return `null`?

Reply:
Updated here https://github.com/elastic/kibana/pull/88880/commits/8a8ae4d6ad73f13327e441f56f7fd805dbb186e7